Userland/Libraries/LibWeb/Layout/GridFormattingContext.cpp

```diff
@@ -461,14 +461,22 @@ void GridFormattingContext::run(Box const& box, LayoutMode)
             auto column_span = child_box.computed_values().grid_column_start().is_span() ? child_box.computed_values().grid_column_start().raw_value() : 1;
             auto row_start = 0;
             auto row_span = child_box.computed_values().grid_row_start().is_span() ? child_box.computed_values().grid_row_start().raw_value() : 1;
-            auto found_unoccupied_cell = false;
+            auto found_unoccupied_area = false;
             for (int row_index = auto_placement_cursor_y; row_index < (int)occupation_grid.size(); row_index++) {
                 for (int column_index = auto_placement_cursor_x; column_index < (int)occupation_grid[0].size(); column_index++) {
-                    if (!occupation_grid[row_index][column_index]) {
-                        found_unoccupied_cell = true;
-                        column_start = column_index;
-                        row_start = row_index;
-                        goto finish;
+                    if (column_span + column_index <= (int)occupation_grid[0].size()) {
```

Comment:
(Not important here) but we generally try to avoid C style casts in favour of the (admittedly more verbose C++ ones). 
(Like `static_cast<int>(val)`)